postcommit.py

```diff
@@ -0,0 +1,642 @@
+                commit['parents'].append({'changeset' :parent_changeset, 'branch' : parent_branch,
+                                          'diff_changes' : ""})
+
+            	if changed_files:
+                	committed_files = list(self.status(parent_changeset, changeset))
+                	committed_files[0] = list(set(committed_files[0]).intersection(set(changed_files)))
+                	committed_files[1] = list(set(committed_files[1]).intersection(set(changed_files)))
+                	committed_files[2] = list(set(committed_files[2]).intersection(set(changed_files)))
+            	else:
+                	committed_files = ([], [], [])
+
+            	commit['files'] = {'Added': self.diffstat(parent_changeset,changeset,committed_files[1]), 'Modified': self.diffstat(parent_changeset,changeset,committed_files[0]), 'Removed': self.diffstat(parent_changeset,changeset,committed_files[2])}
+
+            if len(commit['parents']) == 2:
+                if commit['branch'] == commit['parents'][1]['branch']:
+                    commit['commit_type'] = 3
+                else:
+                    commit['commit_type'] = 2
+            else:
+                commit['commit_type'] = 1
+
+            commits.append(commit)
+        commits.reverse()
+        return commits, newchangesets
+
+    def status(self, changeset1, changeset2):
+        return self.repo.status(changeset1, changeset2, match=None)
+
+    def getAddedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[1]
+
+    def getModifedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[0]
+
+    def getRemovedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[2]
+
+    def getAllFiles(self, changeset):
+
+        return self.repo[changeset].manifest().keys()
+
+    def getChangeContext(self, changeset):
+
+        return self.repo[changeset]
+
+#	***************************************************************************************************************
+#	getDiffOpts :
+#
+#			Provide extra options for diff command
+#
+#       *************************************************************************************************************** 
+    def getDiffOpts(self):
+        options = {'unified': '0', 'ignore_space_change': True, 'ignore_all_space': True}
+        diffoptions = patch.diffopts(self.ui, options)
+        return diffoptions
+
+#	***************************************************************************************************************
+#	getDiffContent :
+#
+#			Show changes between two changeset
+#
+#			@param : changeset1,changeset2
+#			
+#			@return : [{'changed_lines': [1,4,6], 'file': 'test'}]
+#       *************************************************************************************************************** 
+    def getDiffContent(self, changeset1, changeset2):
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "getDiffContent" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)